06-reactjs/components/movie-form.js

```diff
@@ -0,0 +1,32 @@
+var React = require('react');
+
+var MovieForm = React.createClass({
+    /*That clears the form fields when the form is submitted with valid input.*/
+    handleSubmit: function (event) {
+        event.preventDefault();
+        var title = React.findDOMNode(this.refs.title).value.trim();
+        var genre = React.findDOMNode(this.refs.genre).value.trim();
+        var rating = React.findDOMNode(this.refs.rating).value.trim();
+
+        if (!title || !genre || !rating) {
+            return;
+        }
+        this.props.onMovieSubmit({title: title, genre: genre, rating: rating});
```

Comment:
It'll be good to have your props defined within ``propTypes`` (also smells like an ``isRequired`` is needed :P).